caikit/core/data_model/base.py

```diff
@@ -171,6 +181,21 @@ def parse_proto_descriptor(mcs, cls):
             if field.enum_type is not None
         }
 
+        # mapping of all oneofs and the fields that are part of them
+        # NOTE: protobuf makes an interesting use of oneof to wrap types that
```

Comment:
This was important. For _all_ of this oneof logic, I'm excluding oneofs that are used only to wrap `optional` fields.